### PR TITLE
Try intake v2 again

### PR DESCRIPTION
**Pull request by ahuang11**
Supersedes https://github.com/ProjectPythia/landsat-ml-cookbook/pull/22 didn't want to deal with the merge conflicts.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -13,23 +13,42 @@ dependencies:
...
+  # to satisfy dependencies for the sphinx-pythia-theme 2022.3.22 package.
+  - sphinx==4.5.0
+  - sphinxcontrib-applehelp<1.0.7 
+  - sphinxcontrib-devhelp<1.0.5 
+  - sphinxcontrib-htmlhelp<2.0.4 
+  - sphinxcontrib-qthelp<1.0.6 
+  - sphinxcontrib-serializinghtml<1.1.9
+  - pydata-sphinx-theme<=0.8
+  - jsonschema-with-format-nongpl<4.19.1
+  - docutils==0.16
```
Comment (ahuang11) [inline environment.yml]:
```suggestion
```

**Comment by ahuang11** [comment]
@droumis this is ready for review when you have some time; I moved it from the other branch because of the merge conflicts.

**Comment by droumis** [comment]
are you ok with me adding you to the list of authors?

**Comment by ahuang11** [comment]
Yes absolutely; where is the author list?

**Comment by droumis** [comment]
Ok, I did a lot of cleanup and added a lot more contextual narrative framing to the intake notebook. Honestly, I would have much preferred if the intake notebook ends with showing the walker lake satellite image to match the other data ingestion notebook but I have no idea how to select a specific lonlat bbox with the intake approach.  Also the last few cells of the intake notebook just lag indefinitely for me locally.. I'm pushing the work as is anyways to see if it runs and builds eventually on the CI. Hopefully my narrative framing makes sense.. 

I've also updated the toc and reframed the data ingestion options as either Geospatial specific or General purpose (intake), because the intake notebook now uses planetary computer as the source so it didn't make sense to call one of them the 'planetary computer approach' when they both use it. I hope that makes more sense and I hope someone (Martin?) can help improve the intake notebook in the future. 

I also update the spectral clustering notebook to programmatically grab the lake water cluster since it was indeterministic before and ended up being wrong sometimes.

**Comment by ahuang11** [comment]
> Also the last few cells of the intake notebook just lag indefinitely for me locally.. 

It seems to have worked on the website. I think it just takes a long time because it reads the entire stac catalog.

The rest makes sense to me! I'm happy to first merge it so at least it's in a working state, and in the future, it can be enhanced!

Thanks for cleaning it up!

**Comment by ahuang11** [comment]
(I can't approve it since it's my own PR, but feel free to merge when you're ready!)

**Comment by droumis** [comment]
ok, just pushed last commit.. will check the build once it's done then hopefully merge

**Pull request merged**